### PR TITLE
Env var interpolation stop

### PR DESCRIPTION
**Pull request by agnihotrigaurav**
Problem :- variables defined in web are interpolated as shell variables. It creates problem if variable has special characters like $.

Solution :- Stoped interpolation of environment variables defined in web. Public variables has X_ as prefix and protected variables has XXX_ as prefix. Protected variables are already separated as Hidden env  variables. Public variables are separated as well in Public env variables.

Impact:- Pipelines using previous behaviour needs to be changed once PR is merged.

### CODE REVIEW

File: util/environment.go [hunk 1]
```diff
@@ -100,16 +102,41 @@ func (e *Environment) Get(key string) string {
 }
 
 // Export the environment as shell commands for use with Session.Send*
-func (e *Environment) Export() []string {
+func (e *Environment) Export(noInterpolation bool) []string {
 	s := []string{}
```
Comment (nigeldeakin) [inline util/environment.go]: I don't like this because is suggests that the flag `noInterpolation` determines whether environment variables will be interpolated or not. However this is not the case, as can be seen by the code further down: whether or not an anvironment variable is interpolated depends on `noInterpolation || e.contains(key)`. So the variable is misnamed. Please choose a better name and add a comment to explain the logic here.

**Comment by nigeldeakin** [comment]
I tried an environment variable that contained a single quote:
```
export X_QQ='qwe$rty'"'"uio
```
which set `X_QQ" to `X_QQ=qwe$rty'uio`

With the version on master this sets the environment variable within the pipeline to `qwe'uio`, which is what we would expect since it interprets `$rty` as an empty environment variable. 

However with this new branch wercker simply hung.

File: core/pipeline.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ func (pr *PipelineResult) ExportEnvironment(sessionCtx context.Context, sess *Se
 		e.Add("WERCKER_FAILED_STEP_MESSAGE", pr.FailedStepMessage)
 	}
 
-	exit, _, err := sess.SendChecked(sessionCtx, e.Export()...)
+	exit, _, err := sess.SendChecked(sessionCtx, e.Export(false)...)
 	if err != nil {
```
[the remark this answers is not in the document]
Comment (agnihotrigaurav) [reply core/pipeline.go]: Cases which are specific to variables provided by WEB UI are included as per direction given by @gilbode

[the remark this answers is not in the document]
Comment (nigeldeakin) [reply core/pipeline.go]: I don't understand your answer.

**Comment by nigeldeakin** [comment]
I created a branch https://github.com/wercker/wercker/tree/nigels-env-var-test which treats all env vars in the same way. 

This changed `environment.go` and nothing else. See the diff at https://github.com/wercker/wercker/compare/nigels-env-var-test?expand=1#diff-3af917660eeae47c9fcd78d7637cbb6a

I'd like to understand what use case would be broken by the simplistic change, but not with your proposed change. This may need more testing, specifically in the area of passing env vars to steps.

In addition, this PR introduces a bug in which setting an environment variable that contains a single quote causes the pipeline to hang.
